--- linux-4.6/arch/arm/boot/dts/bak/sun7i-a20-cubieboard2_old.dts	2016-05-16 06:43:13.000000000 +0800
+++ linux-4.6/arch/arm/boot/dts/bak/sun7i-a20-cubieboard2_lvds_pwm.dts	2017-08-15 09:37:50.686359000 +0800
@@ -174,6 +174,12 @@
 	};
 };
 
+&pwm {
+      pinctrl-names = "default";
+      pinctrl-0 = <&pwm0_pins_a>, <&pwm1_pins_a>;
+      status = "okay";
+};
+
 &reg_ahci_5v {
 	status = "okay";
 };
